sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -352,21 +352,29 @@ class AccountState(val params: NetworkParams,
   override def getBlockGasLimit: BigInteger = using(getView)(_.getBlockGasLimit)
 
   override def validate(tx: SidechainTypes#SCAT): Try[Unit] = Try {
-    tx.semanticValidity()
+    Try(tx.semanticValidity()) recoverWith {
+      case e: Throwable =>
+        log.debug("Transaction is not semantically valid" , e)
+        Failure(e)
+    }  get
```

Comment:
I expect that we long the error in any case? Why do we need this wrapper + log.debug?

Reply:
Actually, this method is called by Scorex code, that doesn't log caught exceptions. I have moved the recoverWith code so it can wrap all the method and log there all the exceptions.